### PR TITLE
Draft: Add poetry installation support for FastAPI

### PR DESCRIPTION
**Pull request by leehuwuj**
- Add checking is poetry installed (using `poetry --version`).
- Add installing python dependencies (using `poetry install`).
- Add generating index in FastAPI app if the environment was set-up correctly (`poetry run python app/engine/generate.py`).

### CODE REVIEW

File: packages/create-llama/helpers/python.ts [hunk 1]
```diff
@@ -146,7 +148,26 @@ export const installPythonTemplate = async ({
...
 
-  console.log(
-    "\nPython project, dependencies won't be installed automatically.\n",
-  );
+  // install python dependencies
+  if (isPoetryAvailable()) {
+    console.log(
+      `Installing python dependencies using poetry. This may take a while...`,
+    );
+    const is_installed_success = tryPoetryInstall();
```
Comment (marcusschiesser) [inline packages/create-llama/helpers/python.ts:156]: TS is using camel case for variables ;)
```suggestion
    const installSuccessful = tryPoetryInstall();
```

File: packages/create-llama/helpers/poetry.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { execSync } from "child_process";
+import fs from "fs";
+
+export function isPoetryAvailable(): boolean {
+  try {
+    execSync("poetry --version", { stdio: "ignore" });
+    return true;
+  } catch (_) {}
```
Comment (marcusschiesser) [inline packages/create-llama/helpers/poetry.ts:9]: what error handling is used in `callPackageManager`?

Comment (leehuwuj) [reply packages/create-llama/helpers/poetry.ts:9]: sorry, i don't get it. Do you mean that we should handle the error explicitly for this function? about callPackageManager we do not catch the error from/in that

Comment (marcusschiesser) [reply packages/create-llama/helpers/poetry.ts:9]: I mean take a look at `callPackageManager` how they are calling a child process
e.g. they are using `stdio: "inherit"` and checking for the return code which I am not sure is handled here correctly

File: packages/create-llama/helpers/types.ts [hunk 3]
```diff
@@ -23,4 +23,5 @@ export interface InstallTemplateArgs {
   communityProjectPath?: string;
   vectorDb?: TemplateVectorDB;
   externalPort?: number;
+  isPoetryInstall?: boolean;
```
Comment (marcusschiesser) [inline packages/create-llama/helpers/types.ts:26]: generalize this question (also ask it for TS) and rename variable
```suggestion
  installDependencies?: boolean;
```

**Review by marcusschiesser — APPROVED** [review]

File: .changeset/silly-bugs-fold.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+"create-llama": patch
+---
+
+- Add support to install python dependencies
+- Add new create-llama option: `--install-dependencies`:
+  This option allows the user to choose whether to install the dependencies automatically or manually when creating a llama project. This improves the development experience of the tool.
```
Comment (marcusschiesser) [inline .changeset/silly-bugs-fold.md:7]:
```suggestion
Added option to automatically install dependencies (for Python and TS)
```

**Pull request merged**